crema/confidence.py

```diff
@@ -305,27 +339,87 @@ class TdcConfidence(Confidence):
         each level, each as a :py:class:`pandas.DataFrame`
     """
 
-    def __init__(self, psms, score_column=None, desc=None, eval_fdr=0.01):
+    def __init__(
+        self,
+        psms,
+        score_column=None,
+        desc=None,
+        eval_fdr=0.01,
+        pep_fdr_type="psm-peptide",
+    ):
         """Initialize a TdcConfidence object."""
         LOGGER.info(
             "Assigning confidence estimates using target-decoy competition..."
         )
 
         super().__init__(
-            psms=psms, score_column=score_column, desc=desc, eval_fdr=eval_fdr
+            psms=psms,
+            score_column=score_column,
+            desc=desc,
+            eval_fdr=eval_fdr,
+            pep_fdr_type=pep_fdr_type,
         )
 
     def _assign_confidence(self):
         """Assign confidence estimates using target-decoy competition"""
-        df = self.data
         pairing = self.dataset.peptide_pairing
-        pair_col = utils.new_column("pairing", df)
+
+        if pairing == None and self._pep_fdr_type != "psm-only":
+            raise ValueError(
+                "Must provide paired target decoy peptide infomation."
+            )
```

Comment:
I want to subtly (or not so subtly) push people to use psm-peptide since it provides better power. In that case I think leaving an error is better?

I can change the error to indicate that two other peptide-level FDR options exist if you think that would be better.

Reply:
Eh I think it's fine if you really want to push people that way.